scripts/import-wikis-from-local-backup.sh

```diff
@@ -0,0 +1,298 @@
+	# This import script normalizes all database names to be in the form
+	# "wiki_$wiki_id", so if $wiki_id is "eva" then the database is "wiki_eva"
+	#
+	# This command just comments out the old database name
+	sed -i "s/\$mezaCustomDBname/\/\/ \$mezaCustomDBname/g;" "$wiki_install_path/config/preLocalSettings.php"
+
+	# import SQL file
+	# Import database - Ref: https://www.mediawiki.org/wiki/Manual:Restoring_a_wiki_from_backup
+
+	# determine and use most-recent sql file
+	# Ref: http://stackoverflow.com/a/4447795/5103312
+	import_sql_file="$(find $local_wiki_backup/$wiki_id -maxdepth 1 -type f -iname "*.sql" | sort -r | head -n +1)"
+	wiki_db_name="wiki_$wiki_id"
+	echo "For $wiki_db_name: "
+	echo " * dropping if exists"
+	echo " * (re)creating"
+	echo " * importing file at $import_sql_file"
+	mysql -u root "--password=$mysql_root_pass" -e"DROP DATABASE IF EXISTS $wiki_db_name; CREATE DATABASE $wiki_db_name; use $wiki_db_name; SOURCE $import_sql_file;"
+	#
+	# TO-DO: Remove the following rm command; don't remove sql file from backup dir, just use it
+	# rm -rf "$import_sql_file"
+
+
+	# Run update.php. The database you imported may not be up to the same version
+	# as meza, and thus you must update it.
+	#
+	# TO-DO: update.php commented-out for now; may not be necessary
+	#
+	# echo "Running MediaWiki maintenance script \"update.php\""
+	# WIKI="$wiki_id" php "$m_mediawiki/maintenance/update.php" --quick
+
+
+	# if SMW set up yet. On the very first install it will not be.
+	#
+	# TO-DO: SMW maintenance commented-out for now; may not be necessary
+	#
+	# if [ -d "$m_mediawiki/extensions/SemanticMediaWiki" ]; then
+	# 	# Run SMW rebuildData.php
+	# 	# Some documenation says to run this in increments of ~3000 pages, but the most
+	# 	# recent version of http://semantic-mediawiki.org/wiki/Help:RebuildData.php
+	# 	# does not mention that. Attempting without that. If that is required, then
+	# 	# will have to determine a method to test for completion of rebuild, and run it
+	# 	# in a while loop
+	# 	rebuild_exception_log="$m_meza/logs/rebuilddata-exceptions-$wiki_id-.log"
+	# 	echo "Running Semantic MediaWiki maintenance script \"rebuildData.php\""
+	# 	WIKI="$wiki_id" php "$m_mediawiki/extensions/SemanticMediaWiki/maintenance/rebuildData.php" -d 5 -v --ignore-exceptions --exception-log="$rebuild_exception_log"
+
+	# 	# Run runJobs.php
+	# 	# Note that should prob be removed: Daren saw 12k+ jobs in the queue after performing the above steps
+	# 	echo "Running MediaWiki maintenance script \"runJobs.php\""
+	# 	echo "\$wgDisableSearchUpdate = true;" >> "$m_htdocs/wikis/$wiki_id/config/postLocalSettings.php"
+	# 	WIKI="$wiki_id" php "$m_mediawiki/maintenance/runJobs.php" --quick
+	# 	sed -r -i 's/\$wgDisableSearchUpdate = true;//g;' "$m_htdocs/wikis/$wiki_id/config/postLocalSettings.php"
+	# else
+	# 	echo -e "\nSKIPPING SemanticMediaWiki rebuildData.php and runjobs.php (no SMW)"
+	# fi
+
```

Comment:
In `import-wikis.sh` the `elastic-build-index.sh` script is run here. To generalize, it makes sense to move it lower in `import-wikis.sh` too.